### DISCUSSION TITLE
Disable unused fragment error

### DISCUSSION THREAD

**Discussion by danielvanmil** [graphql-java/graphql-java#3077] [Q&A] [answered]
Hello,

Is it maybe possible to disable some validation errors? I'm getting an "unused fragment" error and it would be nice to disable this temporary for now.

**Comment by dondonz** [comment, marked as answer]
Hi @danielvanmil this validation cannot be disabled as it enforces a requirement from the GraphQL specification  https://spec.graphql.org/draft/#sec-Fragments-Must-Be-Used

**Reply by dondonz** [reply]
I was wrong @danielvanmil, **it is possible** to disable validation. However I would put a big red warning label on this! Ignoring the unused fragment rule would make your engine no longer compliant with the GraphQL specification. Ignoring spec validation rules can lead to unpredictable results.

See v18 release notes "Skipping validation rules": https://github.com/graphql-java/graphql-java/releases/tag/v18.0
```
 Predicate<Class<?>> predicate = new Predicate<Class<?>>() {
            @Override
            boolean test(Class<?> aClass) {
                if (aClass == NoUnusedFragments.class) {
                    return false
                }
                return true
            }
        }
        
ExecutionInput.newExecutionInput(query)
                .graphQLContext(["graphql.ParseAndValidate.Predicate": predicate])
                .build()
```

**Reply by danielvanmil** [reply]
Thanks! Could be nice for development when https://github.com/apollographql/apollo-client/issues/10477 might be an issue (but maybe it's my issue).
